Mathlib/Topology/UniformSpace/Basic.lean

```diff
@@ -300,92 +307,101 @@ theorem UniformSpace.Core.ext :
-    UniformSpace α :=
-  ⟨c, isOpen_uniformity⟩
-#align uniform_space.mk' UniformSpace.mk'
+#noalign uniform_space.mk' -- Can't be a `match_pattern`, so not useful anymore
 
-/-- Construct a `UniformSpace` from a `UniformSpace.Core`. -/
-@[reducible]
-def UniformSpace.ofCore {α : Type u} (u : UniformSpace.Core α) : UniformSpace α where
-  toCore := u
-  toTopologicalSpace := u.toTopologicalSpace
-  isOpen_uniformity _ := Iff.rfl
-#align uniform_space.of_core UniformSpace.ofCore
+/-- The uniformity is a filter on α × α (inferred from an ambient uniform space
+  structure on α). -/
+def uniformity (α : Type u) [UniformSpace α] : Filter (α × α) :=
+  @UniformSpace.uniformity α _
+#align uniformity uniformity
+
+/-- Notation for the uniformity filter with respect to a non-standard `UniformSpace` instance. -/
+scoped[Uniformity] notation "𝓤[" u "]" => @uniformity _ u
+
+@[inherit_doc] -- Porting note (#11215): TODO: should we drop the `uniformity` def?
+scoped[Uniformity] notation "𝓤" => uniformity
 
 /-- Construct a `UniformSpace` from a `u : UniformSpace.Core` and a `TopologicalSpace` structure
 that is equal to `u.toTopologicalSpace`. -/
 @[reducible]
 def UniformSpace.ofCoreEq {α : Type u} (u : UniformSpace.Core α) (t : TopologicalSpace α)
     (h : t = u.toTopologicalSpace) : UniformSpace α where
-  toCore := u
+  __ := u
   toTopologicalSpace := t
-  isOpen_uniformity _ := h.symm ▸ Iff.rfl
+  nhds_eq_comap_uniformity x := by rw [h, u.nhds_toTopologicalSpace]
 #align uniform_space.of_core_eq UniformSpace.ofCoreEq
 
+/-- Construct a `UniformSpace` from a `UniformSpace.Core`. -/
+@[reducible]
+def UniformSpace.ofCore {α : Type u} (u : UniformSpace.Core α) : UniformSpace α :=
+  .ofCoreEq u _ rfl
+#align uniform_space.of_core UniformSpace.ofCore
+
+/-- Construct a `UniformSpace.core` from a `UniformSpace`. -/
+@[reducible]
+def UniformSpace.toCore (u : UniformSpace α) : UniformSpace.Core α where
+  __ := u
+  refl := by
+    rintro U hU ⟨x, y⟩ (rfl : x = y)
+    have : Prod.mk x ⁻¹' U ∈ 𝓝 x := by
+      rw [UniformSpace.nhds_eq_comap_uniformity]
+      exact preimage_mem_comap hU
+    convert mem_of_mem_nhds this
+
 theorem UniformSpace.toCore_toTopologicalSpace (u : UniformSpace α) :
     u.toCore.toTopologicalSpace = u.toTopologicalSpace :=
-  TopologicalSpace.ext <| funext fun s => propext (UniformSpace.isOpen_uniformity s).symm
+  eq_of_nhds_eq_nhds fun a ↦ by rw [u.nhds_eq_comap_uniformity, u.toCore.nhds_toTopologicalSpace]
 #align uniform_space.to_core_to_topological_space UniformSpace.toCore_toTopologicalSpace
 
--- Porting note (#11215): TODO: use this as the main definition?
-/-- An alternative constructor for `UniformSpace` that takes the proof of `nhds_eq_comap_uniformity`
-as an argument. -/
-@[reducible]
-def UniformSpace.ofNhdsEqComap (u : UniformSpace.Core α) (t : TopologicalSpace α)
+/-- The main constructor used to have a different assumption. -/
```

Comment:
This docstring looks useless to me if one doesn't already know what this is about, could you expand it a little bit?

Reply:
```suggestion
/-- Build a `UniformSpace` from a `UniformSpace.Core` and a compatible topology.
Use `UnifiormSpace.mk` instead to avoid proving
the unnecessary assumption `UniformSpace.Core.refl`.

The main constructor used to use a different compatibility assumption.
This definition was created as a step towards porting to a new definition.
Now the main definition is ported,
so this constructor will be removed in a few months. -/
```